dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/AkkaHttpServiceGateway.scala

```diff
@@ -30,15 +30,21 @@ import scala.collection.immutable
 import scala.concurrent.duration._
 import scala.concurrent.{ Await, Future }
 
-class AkkaHttpServiceGatewayFactory @Inject() (lifecycle: ApplicationLifecycle, config: ServiceGatewayConfig,
-                                               @Named("serviceRegistryActor") registry: ActorRef)(implicit actorSystem: ActorSystem, mat: Materializer) {
+class AkkaHttpServiceGatewayFactory @Inject() (
+  coordinatedShutdown:                     CoordinatedShutdown,
+  config:                                  ServiceGatewayConfig,
+  @Named("serviceRegistryActor") registry: ActorRef
+)(implicit
+  actorSystem: ActorSystem,
+  mat: Materializer
+) {
```

Comment:
Formatting seems pretty difficult to read. Is this scalariform's work?

I wonder if it will let us break the annotation onto a separate line.

Reply:
I found a way to cheat:

```diff
diff --git i/build.sbt w/build.sbt
index c99d097a..02e9ac6e 100644
--- i/build.sbt
+++ w/build.sbt
@@ -165,6 +165,7 @@ def formattingPreferences = {
     .setPreference(RewriteArrowSymbols, false)
     .setPreference(AlignParameters, true)
     .setPreference(AlignSingleLineCaseStatements, true)
+    .setPreference(AllowParamGroupsOnNewlines, true)
     .setPreference(SpacesAroundMultiImports, true)
     .setPreference(DanglingCloseParenthesis, Force)
     .setPreference(AlignArguments, false)
diff --git i/dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/AkkaHttpServiceGateway.scala w/dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/AkkaHttpServiceGateway.scala
index 95c650ab..3bc7f29a 100644
--- i/dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/AkkaHttpServiceGateway.scala
+++ w/dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/gateway/AkkaHttpServiceGateway.scala
@@ -30,14 +30,9 @@ import scala.collection.immutable
 import scala.concurrent.duration._
 import scala.concurrent.{ Await, Future }
 
-class AkkaHttpServiceGatewayFactory @Inject() (
-  coordinatedShutdown:                     CoordinatedShutdown,
-  config:                                  ServiceGatewayConfig,
-  @Named("serviceRegistryActor") registry: ActorRef
-)(implicit
-  actorSystem: ActorSystem,
-  mat: Materializer
-) {
+class AkkaHttpServiceGatewayFactory @Inject() (coordinatedShutdown: CoordinatedShutdown, config: ServiceGatewayConfig)
+  (@Named("serviceRegistryActor") registry: ActorRef)
+  (implicit actorSystem: ActorSystem, mat: Materializer) {
 
   def start(): InetSocketAddress = {
     new AkkaHttpServiceGateway(coordinatedShutdown, config, registry).address
```

WDYT, better or worse?